Added VTOL Takeoff navigation mode

RomanBapst opened the pull request:
**Describe problem solved by this pull request**
Please have a look at the corresponding pull request on QGroundControl:
https://github.com/mavlink/qgroundcontrol/pull/10123

sfuhrer commented on src/lib/events/enums.json:
```diff
@@ -239,6 +239,10 @@
                         },
+                        "15": {
+                            "name": "auto_vtol_takeoff",
+                            "description": "Vtol takeoff"
```
```suggestion
                            "description": "VTOL Takeoff"
```

sfuhrer commented on src/modules/navigator/navigator.h:
```diff
@@ -78,14 +79,15 @@
 #include <uORB/topics/vehicle_land_detected.h>
 #include <uORB/topics/vehicle_local_position.h>
 #include <uORB/topics/vehicle_status.h>
+#include <uORB/topics/wind.h>
```
don't think wind is needed for the VTOL takeoff, is it?

sfuhrer commented on src/modules/navigator/navigator_main.cpp:
```diff
@@ -455,6 +458,16 @@ Navigator::run()
+				_vtol_takeoff.setLoiterLocation(matrix::Vector2d(cmd.param5, cmd.param6));
+
+				// loiter height is the height above takeoff altitude at which the vehicle will establish on a loiter circle
+				_vtol_takeoff.setLoiterHeight(cmd.param1);
```
That's not aligned with current mavlink specs right? Do you plan to open a mavlink PR for it, and meanwhile use VTO_LOITER_ALT instead?

RomanBapst replied:
@sfuhrer Yes, exactly. The value is currently not used, we still stick to the parameter. In order to use the value from the command we need to extend the mavlink specs and also add some UI in QGC for specifying the altitude.

sfuhrer commented on src/modules/navigator/vtol_takeoff.cpp:
```diff
@@ -0,0 +1,178 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2021 PX4 Development Team. All rights reserved.
```
2022

sfuhrer commented on src/modules/navigator/vtol_takeoff.cpp:
```diff
@@ -0,0 +1,178 @@
+		case vtol_takeoff_state::TRANSITION: {
+				position_setpoint_triplet_s *pos_sp_triplet = _navigator->get_position_setpoint_triplet();
+
+				if (pos_sp_triplet->current.valid && pos_sp_triplet->current.type == position_setpoint_s::SETPOINT_TYPE_LOITER) {
```
In which case is that condition valid (type LOITER) and when the other?

sfuhrer commented on src/modules/navigator/vtol_takeoff.cpp:
```diff
@@ -0,0 +1,178 @@
+				_mission_item.lon = pos_sp_triplet->current.lon;
+
+
+				//publish_navigator_mission_item(); // for logging
```
do we need publish_navigator_mission_item() or not?

RomanBapst commented:
@sfuhrer Thanks for the review, will address shortly

RomanBapst commented:
@dagar Would you mind having a quick look at this as well?

RomanBapst commented:
@sfuhrer I've addressed your comments which also lead to more cleanup in the vtol takeoff navigation mode. We had some residual stuff in there that was not needed according to my testing.

sfuhrer reviewed (approved):
All my comments were addressed and SITL testing passed.
